src/main/java/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,63 @@
+package bookstore.controller;
+
+import bookstore.dto.BookDtoWithoutCategoryIds;
+import bookstore.dto.category.CategoryDto;
+import bookstore.dto.category.CreateCategoryRequestDto;
+import bookstore.service.book.BookService;
+import bookstore.service.category.CategoryService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("api/category")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping
+    public CategoryDto createCategory(CreateCategoryRequestDto requestDto) {
+        return categoryService.save(requestDto);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    public List<CategoryDto> getAll(Pageable pageable) {
+        return categoryService.findAll(pageable);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    public CategoryDto getCategoryById(Long id) {
+        return categoryService.getById(id);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping
+    public CategoryDto updateCategory(Long id, CreateCategoryRequestDto requestDto) {
+        return categoryService.update(id,requestDto);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping
+    void deleteCategory(Long id) {
+        categoryService.deleteById(id);
+    }
+
+    @DeleteMapping("/{id}")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```

Comment:
```suggestion
    @PreAuthorize("hasRole('ADMIN')")
```
There is some diff between methods hasRole and hasAuthority 
https://stackoverflow.com/questions/19525380/difference-between-role-and-grantedauthority-in-spring-security